lib/winrm/shells/power_shell.rb

```diff
@@ -34,12 +34,16 @@ def finalize(connection_opts, transport, shell_id)
         end
 
         def close_shell(connection_opts, transport, shell_id)
+          return false unless Thread.current.alive?
```

Comment:
Just curious, if `Thread.current` is not alive, how could this code be executing?

Reply:
The thread is generally in asleep in my testing results. If it’s dead, I don’t think there’s much recourse that won’t potentially cause additional side effects. Maybe we rescue instead of returning.

Reply:
It strikes me as super odd that the thread would be sleeping if the code is actually running. In your testing, are you examining the thread status of `Thread.current` or are you observing the thread's status from a different thread.